core/src/main/java/org/apache/gravitino/listener/ModelEventDispatcher.java

```diff
@@ -0,0 +1,268 @@
+            ident,
+            registerModelRequest,
+            linkModelVersionRequest);
+    eventBus.dispatchEvent(registerAndLinkModelPreEvent);
+    try {
+      // TODO: ModelEvent
+      return dispatcher.registerModel(ident, uri, aliases, comment, properties);
+    } catch (Exception e) {
+      throw e;
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Model getModel(NameIdentifier ident) throws NoSuchModelException {
+    eventBus.dispatchEvent(new GetModelPreEvent(PrincipalUtils.getCurrentUserName(), ident));
+    try {
+      Model model = dispatcher.getModel(ident);
+      // TODO: ModelEvent
+      return model;
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public boolean deleteModel(NameIdentifier ident) {
+    eventBus.dispatchEvent(new DeleteModelPreEvent(PrincipalUtils.getCurrentUserName(), ident));
+    try {
+      // TODO: ModelEvent
+      return dispatcher.deleteModel(ident);
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public NameIdentifier[] listModels(Namespace namespace) throws NoSuchSchemaException {
+    eventBus.dispatchEvent(new ListModelPreEvent(PrincipalUtils.getCurrentUserName(), namespace));
+    try {
+      NameIdentifier[] models = dispatcher.listModels(namespace);
+      // TODO: ModelEvent
+      return models;
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void linkModelVersion(
+      NameIdentifier ident,
+      String uri,
+      String[] aliases,
+      String comment,
+      Map<String, String> properties)
+      throws NoSuchModelException, ModelVersionAliasesAlreadyExistException {
+    ModelVersionInfo linkModelRequest = new ModelVersionInfo(uri, comment, properties, aliases);
+    eventBus.dispatchEvent(
+        new LinkModelVersionPreEvent(PrincipalUtils.getCurrentUserName(), ident, linkModelRequest));
+    try {
+      dispatcher.linkModelVersion(ident, uri, aliases, comment, properties);
+      // TODO: ModelEvent
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ModelVersion getModelVersion(NameIdentifier ident, int version)
+      throws NoSuchModelVersionException {
+    eventBus.dispatchEvent(new GetModelVersionPreEvent(PrincipalUtils.getCurrentUserName(), ident));
+    try {
+      // TODO: ModelEvent
+      return dispatcher.getModelVersion(ident, version);
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ModelVersion getModelVersion(NameIdentifier ident, String alias)
+      throws NoSuchModelVersionException {
+    eventBus.dispatchEvent(new GetModelVersionPreEvent(PrincipalUtils.getCurrentUserName(), ident));
```

Comment:
could you add `alias` and `version` to `GetModelVersionPreEvent` like `DeleteModelVersionPreEvent`?